SMelidoni/personal-finance-manager
# feature/secure-route-access

## Description of Changes

This PR introduces changes related to the authentication flow of the application. It implements Redux for state management, particularly for tracking the authentication state across the application. Additionally, it introduces the usage of localStorage for persisting the authentication state across browser sessions. To ensure the user is authenticated when accessing certain routes, a PrivateRoute component was created. A server-side endpoint '/auth/check' is also used to verify if the user is authenticated.

## Type of Change

- Bug Fix
- Breaking Change
- Documentation Update
- Other

## Describe the problem this PR solves and your solution.

The problem was that the authentication state was not being properly managed and persisted across the application. The solution was to introduce Redux for state management, use localStorage for persistence, create a PrivateRoute component for protected routes, and utilise a server-side endpoint '/auth/check' for verifying the user's authentication state.

## Testing Steps

1. Login with valid credentials.
2. Navigate to a protected route e.g. `/home`, you should be able to access it.
3. Refresh the page. You should still be on the protected route and not redirected to the login page.
4. Log out and try to access the protected route. You should be redirected to the login page.

## Screenshots

_Add any images that will help this PR to be reviewed_

N/A

## Checklist
